Source/WAutoViewLayoutVC.swift

```diff
@@ -0,0 +1,220 @@
+        attributes?.forEach { layoutAttribute in
+            if (layoutAttribute.frame.origin.y >= maxY) {
+                leftMargin = sectionInset.left
+            }
+
+            layoutAttribute.frame.origin.x = leftMargin
+
+            leftMargin += layoutAttribute.frame.width + minimumInteritemSpacing
+            maxY = max(layoutAttribute.frame.maxY , maxY)
+        }
+
+        return attributes
+    }
+}
+
+public class WAutoViewLayoutVC: UIViewController {
+    public var collectionView = UICollectionView(frame: CGRectZero, collectionViewLayout: UICollectionViewFlowLayout())
+    public var flowLayout = UICollectionViewFlowLayout()
+
+    public var alignment: xAlignment = .Center {
+        didSet {
+            if (alignment != oldValue) {
+                switch alignment {
+                case .Right:
+                    print("Only left and centered are currently supported. Defaulting to centered...")
+                    alignment = .Center
+                    flowLayout = UICollectionViewFlowLayout()
+                case .Left:
+                    flowLayout = WLeftAlignedCollectionViewFlowLayout()
+                case .Center:
+                    flowLayout = UICollectionViewFlowLayout()
+                }
+
+                updateCollectionView()
+                refreshAutoViewLayout()
+            }
+        }
+    }
+
+    public var views: [UIView] = [] {
+        didSet {
+            refreshAutoViewLayout()
+        }
+    }
+
+    public var leftSpacing: CGFloat = 5 {
+        didSet {
+            updateCollectionView()
+        }
+    }
+    public var topSpacing: CGFloat = 5 {
+        didSet {
+            updateCollectionView()
+        }
+    }
+    public var rightSpacing: CGFloat = 5 {
+        didSet {
+            updateCollectionView()
+        }
+    }
+    public var bottomSpacing: CGFloat = 5 {
+        didSet {
+            updateCollectionView()
+        }
+    }
+
+    public var fittedHeight: CGFloat {
+        get {
+            return collectionView.collectionViewLayout.collectionViewContentSize().height
+        }
+    }
+
+    public class func estimatedFittedHeight(views: [UIView], constrainedWidth: CGFloat, leftSpacing: CGFloat = 5, topSpacing: CGFloat = 5, rightSpacing: CGFloat = 5, bottomSpacing: CGFloat = 5) -> CGFloat {
+        let autoViewLayoutVC = WAutoViewLayoutVC(width: constrainedWidth)
+        autoViewLayoutVC.leftSpacing = leftSpacing
+        autoViewLayoutVC.topSpacing = topSpacing
+        autoViewLayoutVC.rightSpacing = rightSpacing
+        autoViewLayoutVC.bottomSpacing = bottomSpacing
+
+        autoViewLayoutVC.views = views
+
+        return autoViewLayoutVC.fittedHeight
+    }
+
+    let reuseIdentifier = "cell"
+
+    /// Use so that the size of the collection view is correct
+    public convenience init(width: CGFloat) {
+        self.init()
+        updateWidth(width)
+    }
+
+    /// Use if the height of the collection view is not set on the init
```

Comment:
[minor] since this is the helper text, should we make it more descriptive of what this will do? i.e. `If the height of the collection view is not set during init, call this method to update the width.`?